discovery/openstack/openstack.go

```diff
@@ -91,12 +94,20 @@ func (c *SDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	if err != nil {
 		return err
 	}
+
+	switch c.Availability {
+	case string(gophercloud.AvailabilityPublic), string(gophercloud.AvailabilityInternal), string(gophercloud.AvailabilityAdmin):
+	default:
+		return fmt.Errorf("unknown availability %s, must be one of %s, %s or %s", c.Availability, gophercloud.AvailabilityAdmin, gophercloud.AvailabilityInternal, gophercloud.AvailabilityPublic)
```

Comment:
@brian-brazil or did you mean you'd like this more:
```suggestion
	case "public", "internal", "admin":
	default:
		return fmt.Errorf("unknown availability %s, must be one of admin, internal or public", c.Availability)
```

Reply:
Yes, like that. All the indirection makes it difficult to read - I don't want to have to check an entire other repository to see what the strings are.

Reply:
Much quicker to read now, thank you for the suggestion.